File: product-service/src/main/kotlin/com/lcaohoanq/productservice/repository/CartItemRepository.java
package com.lcaohoanq.productservice.repository;


import com.lcaohoanq.productservice.domain.cart.CartItem;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CartItemRepository extends JpaRepository<CartItem, Long> {
    
    List<CartItem> findByCartId(Long cartId);

    Optional<CartItem> findByCartIdAndProductId(long cartId, long productId);
    
}

File: product-service/src/main/kotlin/com/lcaohoanq/productservice/repository/CartItemRepository.kt
package com.lcaohoanq.productservice.repository

import com.lcaohoanq.productservice.domain.cart.CartItem
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*


interface CartItemRepository : JpaRepository<CartItem, Long> {
    fun findByCartId(cartId: Long): List<CartItem>

    fun findByCartIdAndProductId(cartId: Long, productId: Long): Optional<CartItem>
}
